Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1062 @@
+            $isMinTLSVersionSetOnCustomDomainPostRemediation = $_.MinimumTlsVersion
+            
+            $frontdoorEndpoint | Add-Member -NotePropertyName isMinTLSVersionSetOnCustomDomainPostRemediation -NotePropertyValue $isMinTLSVersionSetOnCustomDomainPostRemediation
+            $frontdoorEndpoint | Add-Member -NotePropertyName PrevMinimumTlsVersion -NotePropertyValue $PrevMinimumTlsVersion
+
+            Write-Host "Setting minimum required TLS Version for  Front Door :  Resource Group Name: [$($resourceGroupName)], Resource Name: [$($resourceName)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            
+            if (-not [System.Convert]::ToBoolean($isMinTlsVersionSetOnCustomDomain))
+            {
+                try
+                {
+                    Write-Host "Setting minimum required TLS version on custom domain : $[$DomainName]" -ForegroundColor $([Constants]::MessageType.Info)
+                    Write-Host $([Constants]::SingleDashLine)
+                    
+                    $secrets =  Get-AzFrontDoorCdnSecret -ResourceGroupName $resourceGroupName -ProfileName $resourceName
+                    $secretObj = $null
+
+                    foreach($secret in $secrets)
+                    {
+                        if($secret.Name.Contains($DomainName))
+                        {
+                            $secretObj = $secret
+                        }
+                    }
+                   
+                    $secretResoure = New-AzFrontDoorCdnResourceReferenceObject -Id $secretObj.Id
+                    $updateTlsSetting = New-AzFrontDoorCdnCustomDomainTlsSettingParametersObject -CertificateType $CertificateType -MinimumTlsVersion $requiredMinTLSVersion -Secret $secretResoure
+                    $resource = Update-AzFrontDoorCdnCustomDomain -ResourceGroupName $resourceGroupName -ProfileName $resourceName -CustomDomainName $DomainName -TlsSetting $updateTlsSetting
+
+                    if($resource)
+                    {
+                        $frontdoorEndpoint.isMinTLSVersionSetOnCustomDomainPostRemediation = $true
+                        $frontdoorEndpoint.minimumTlsVersion = $requiredMinTLSVersion
+                        $customDomainsRemediated += $frontdoorEndpoint
+                        Write-Host "Minimum required TLS version for Front Door has been set successfully for " $[$DomainName]  -ForegroundColor $([Constants]::MessageType.Update)
+                        Write-Host $([Constants]::SingleDashLine)
+                    }
+                    else
+                    {
+                        $frontDoorsEPSkipped += $frontdoorEndpoint
+                        $logResource = @{}
+                        $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+                        $logResource.Add("ResourceName",($_.ResourceName))
```

Comment:
This resource name will have, front door name or endpoint name.

Reply:
Added domain name to logs